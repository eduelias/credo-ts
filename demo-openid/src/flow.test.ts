import { Holder } from './Holder'
import { Issuer, credentialsSupported } from './Issuer'
import { Verifier, presentationDefinitions } from './Verifier'

describe('Testing the happy path', () => {
  it('should run the happy path', async () => {
    const credential = credentialsSupported[0]
    const issuer = await Issuer.build()
    const holder = await Holder.build()
    const verifier = await Verifier.build()

    // Issue, resolve and store credential
    const credentialOffer = await issuer.createCredentialOffer([credential.id])
    const resolvedCredentialOffer = await holder.resolveCredentialOffer(credentialOffer)
    const accepted = await holder.requestAndStoreCredentials(
      resolvedCredentialOffer,
      resolvedCredentialOffer.offeredCredentials.map((credential) => credential.id)
    )

    // Request presentation/ present and verify
    const proofRequest = await verifier.createProofRequest(presentationDefinitions[0])
    const resolvedProofRequest = await holder.resolveProofRequest(proofRequest)
    const resolvedPresentationRequest = await holder.acceptPresentationRequest(resolvedProofRequest)

    expect(resolvedPresentationRequest).toMatchSnapshot()
    expect(resolvedPresentationRequest.status).toBe(200)
  })
})
