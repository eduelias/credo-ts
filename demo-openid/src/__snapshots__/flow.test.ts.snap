// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Testing the happy path should run the happy path 1`] = `
{
  "body": "<!DOCTYPE html>
<html lang="en">
<head>
<meta charset="utf-8">
<title>Error</title>
</head>
<body>
<pre>Cannot POST /siop/6978131a-5558-4df5-b78c-4512679114a8/authorize</pre>
</body>
</html>
",
  "status": 404,
}
`;
